dtp/src/main.rs

```diff
@@ -77,8 +76,19 @@ async fn main() -> Result<()> {
             // get the .env
             dotenv::from_path("./dtp/.env").expect("Failed to get env variables");
 
-            let (counter_address, load_address, multicall_address, fund_contract_addr) =
-                get_contract_addresses_from_env().await?;
+            // init logger for debugging
+            env_logger::init();
+
+            // get the env variables
+            let (
+                counter_address,
+                load_address,
+                multicall_address,
+                fund_contract_addr,
+                chain_id,
```

Comment:
You could just set `chain_id` from a single call to the rpc rather than force the user to add it as an environment variable. The chain id is going to have to match what the rpc provides anyway. The big improvement from prior implementation is to not make the call on every signed transaction.

Reply:
Fair enough. I will do the required changes.

Reply:
[commit](https://github.com/abhi3700/domain-transaction-producer/pull/5/commits/45f07ad16d8f96946129a30f1a0da2e076e6c0e6)